### PR TITLE
add method to handle channels as time

### PR DESCRIPTION
**Pull request by jdroenner**

### CODE REVIEW

File: src/util/gdal_timesnap.cpp [hunk 1]
```diff
@@ -214,6 +214,25 @@ GDALTimesnap::GDALDataLoadingInfo GDALTimesnap::getDataLoadingInfo(Json::Value d
...
         Log::debug(concat("getDataLoadingInfo: resulting time fileName: ", fileName.c_str()));
 
+    } else if (
+        datasetJson.isMember("channel_start_time_list")
+        && datasetJson["channel_start_time_list"].isArray()
+    ) {
+
+        Log::debug(concat("getDataLoadingInfo: using channels as time"));
+
+        auto time_channel = 0;
```
Comment (ChristianBeilschmidt) [inline src/util/gdal_timesnap.cpp:224]: What happens if time is too small and channel stays zero?

File: src/util/gdal_timesnap.cpp [hunk 2]
```diff
@@ -214,6 +214,25 @@ GDALTimesnap::GDALDataLoadingInfo GDALTimesnap::getDataLoadingInfo(Json::Value d
...
+        auto channel_time_strings = datasetJson.get("channel_start_time_list", Json::Value(Json::ValueType::arrayValue));
+        for (auto &time_string : channel_time_strings) {
+            auto channel_time = timeParser->parse(time_string.asString());
```
Comment (ChristianBeilschmidt) [inline src/util/gdal_timesnap.cpp]:
```suggestion
        const auto channel_time_strings = datasetJson.get("channel_start_time_list", Json::Value(Json::ValueType::arrayValue));
        for (const auto &time_string : channel_time_strings) {
            const auto channel_time = timeParser->parse(time_string.asString());
```

**Review by ChristianBeilschmidt — COMMENTED** [review]
It looks somewhat fine, although a very hacky solution.
The system ignores the previously set channel.

How does the system handle channel = 0?

File: src/util/gdal_timesnap.cpp (hunk 1 above)
Comment (jdroenner) [reply src/util/gdal_timesnap.cpp:224]: i have added a NoRasterForGivenTimeException

File: src/util/gdal_timesnap.cpp (hunk 2 above)
Comment (jdroenner) [reply src/util/gdal_timesnap.cpp]: done

**Pull request merged**